### PR TITLE
test: new test cases created for inactive customers report and customer credit balance report

### PR DESCRIPTION
**Pull request by vinod-8848digital**
### Title: Test Cases for Customer Reports – Inactive Customers and Credit Balance
# Description:
### Test Summary
 - This suite enhances automated validation for customer-related reports in the ERPNext Selling module by focusing on inactive customers and credit balances. It ensures that key reports handle filter logic, missing parameters, and customer credit calculations correctly, even in edge cases. The tests simulate realistic business scenarios involving Sales Orders and Invoices, enforcing business logic such as inactive customer tracking and credit usage accuracy.

- **TC_S_208**: Validates the "Inactive Customers" report behavior by simulating a customer who placed a Sales Order 2 days ago. Creates a customer and sales order dated 2 days prior. Executes the inactive_customers report using days_since_last_order  2.  It should ignores the last two days records and prior to that, Confirms that the correct customer appears in the results with expected customer, territory, and group labels. Asserts order count and billing amounts are accurately shown.

 - **TC_S_209**: Ensures the "Inactive Customers" report raises a validation error if required filters (e.g. days_since_last_order) are not provided. Executes the report without passing any filters. Expects a frappe.ValidationError with a meaningful message indicating that the days_since_last_order filter is mandatory. Ensures graceful failure to prevent silent data issues or crashes.

 - **TC_S_210**: Tests the accuracy of the "Customer Credit Balance" report by simulating a customer with a predefined credit limit and a submitted sales invoice. Creates a customer with a credit limit of 100,000. Creates and submits a Sales Invoice for 100. Runs the customer_credit_balance report filtered by company and customer. Asserts the report returns correct values for customer name, credit limit, invoiced amount, and available credit.

### Scenarios Covered

**Input Validations**

 - Filter enforcement for inactive customer reports.

 - Validation of credit limits and available balances.

 - Business Rule Enforcement:

 - Accurate detection of customer activity status based on Sales Orders.

 - Proper calculation of credit usage after invoicing.

**Edge Case Coverage**
- Missing filters raise controlled exceptions.
- Accurate results even when orders or invoices are backdated.
- Data Integrity & Reporting Accuracy:
- Correct linkage of customers, items, invoices, and order dates.
- Accurate computation of available credit from credit limits and sales invoices.

**API Response Handling**

 - N/A

**Technology**

 - Python unittest framework
 - Frappe/ERPNext testing utilities

**Linked Feature/Bug**
N/A

**Issue Tracker ID**
N/A

### CODE REVIEW

File: erpnext/selling/report/inactive_customers/test_inactive_customers.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+					self.assertEqual(row[1], "_Test Inactive Customer")
+					self.assertEqual(row[2], "_Test Territory")
+					self.assertEqual(row[3], "_Test Customer Group")
+					self.assertEqual(row[4], 1)
+					self.assertEqual(row[5], 1000)
+					self.assertEqual(row[6], 1000)
+
+	def test_validate_filters_TC_S_209(self):
+		with self.assertRaises(frappe.ValidationError) as context:
+			execute()
```
Comment (aasif-patel) [inline erpnext/selling/report/inactive_customers/test_inactive_customers.py]: @vinod-8848digital use assertRaises with msg parameter, check stander test for ref

**Review by aasif-patel — CHANGES_REQUESTED** [review]

File: erpnext/selling/report/inactive_customers/test_inactive_customers.py (hunk 1 above)
Comment (vinod-8848digital) [reply erpnext/selling/report/inactive_customers/test_inactive_customers.py]: @aasif-patel test case is updated with standard assertRaise msg parameter,

**Review by aasif-patel — APPROVED** [review]

**Comment by swathibollineni2** [comment]
@vinod-8848digital All test cases passed
